pkg/openthread/contrib/platform_radio.c

```diff
@@ -213,17 +238,17 @@ void otPlatRadioSetExtendedAddress(otInstance *aInstance, const otExtAddress *aE
 {
     (void)aInstance;
     DEBUG("openthread: otPlatRadioSetExtendedAddress\n");
-    char reversed_addr[IEEE802154_LONG_ADDRESS_LEN];
+    uint8_t reversed_addr[IEEE802154_LONG_ADDRESS_LEN];
     for (unsigned i = 0; i < IEEE802154_LONG_ADDRESS_LEN; i++) {
-        reversed_addr[i] = (uint8_t) ((uint8_t *)aExtAddress)[IEEE802154_LONG_ADDRESS_LEN - 1 - i];
+        reversed_addr[i] = aExtAddress->m8[IEEE802154_LONG_ADDRESS_LEN - 1 - i];
```

Comment:
This is network vs host byte order, right? If so, `reversed_addr` is a misleading name. (You could also take a look at the `eui64_t` type and see if it would be handy here.) 

If the host byte order happens to be network byte order, this would not work. Maybe `_set_long_addr()` could be changed to always expect network byte order? (Or even `eui64_t`?)

Reply:
OpenThread uses IEEE addresses in little endian (see [OpenThread radio API](https://github.com/openthread/openthread/blob/295ce1125ec6b0b38a2de90d0a6e54d010a5a061/include/openthread/platform/radio.h#L341))

So we need to reverse in order to write it in the hardware. Maybe we could choose a better name for this variable or add comments?